packages/framework-provider-aws/src/library/events-searcher-adapter.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  BoosterConfig,
+  EventEnvelope,
+  EventFilter,
+  EventInterface,
+  EventSearchResponse,
+  Logger,
+  UUID,
+} from '@boostercloud/framework-types'
+import { DynamoDB } from 'aws-sdk'
+import { dynamoDbBatchGetLimit, eventsStoreAttributes } from '../constants'
+import { DocumentClient } from 'aws-sdk/clients/dynamodb'
+import { decodeEventStoreSortingKey, partitionKeyForEvent, partitionKeyForIndexByEntity } from './keys-helper'
+
+export async function searchEvents(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  filters: EventFilter
+): Promise<Array<EventSearchResponse>> {
+  logger.debug('Initiating an events search. Filters: ', filters)
+  const timeFilterQuery = buildSearchEventsTimeQuery(filters.from, filters.to)
+  const eventEnvelopes = await executeSearch(dynamoDB, config, logger, filters, timeFilterQuery)
+
+  logger.debug('Events search result: ', eventEnvelopes)
+  return convertToSearchResult(eventEnvelopes)
+}
+
+interface TimeQueryData {
+  expression: string
+  attributeValues: Record<string, string>
+}
+
+function buildSearchEventsTimeQuery(from?: string, to?: string): TimeQueryData {
+  let timeQueryData: TimeQueryData = {
+    expression: '',
+    attributeValues: {},
+  }
+  if (from && to) {
+    timeQueryData = {
+      expression: ` AND ${eventsStoreAttributes.sortKey} BETWEEN :fromTime AND :toTime`,
+      attributeValues: {
+        ':fromTime': from,
+        ':toTime': to,
+      },
+    }
+  } else if (from) {
+    timeQueryData = {
+      expression: ` AND ${eventsStoreAttributes.sortKey} >= :fromTime`,
+      attributeValues: { ':fromTime': from },
+    }
+  } else if (to) {
+    timeQueryData = {
+      expression: ` AND ${eventsStoreAttributes.sortKey} <= :toTime`,
+      attributeValues: { ':toTime': to },
+    }
+  }
+  return timeQueryData
+}
+
+async function executeSearch(
```

Comment:
Depending on the type of query (by entity, by entity with an ID, by type), we need to query different indexes